app/models/topic.rb

```diff
@@ -1370,6 +1370,16 @@ def is_category_topic?
     @is_category_topic ||= Category.exists?(topic_id: self.id.to_i)
   end
 
+  def reset_bumped_at
+    post = ordered_posts.where(
+      user_deleted: false,
+      hidden: false,
+      post_type: Topic.visible_post_types
+    ).last
+
+    update_attribute(:bumped_at, post.created_at)
```

Comment:
Since we're not checking for the return value, we would want to use `update!` here so it doesn't fail silently.